docs/format.rst

```diff
@@ -53,31 +57,147 @@ File Structure
+migration to a new data format so that old readers can still read new data while
+a migration is in progress.
+
+Encodings are divided into "field encodings" and "array encodings".  Field encodings
+are consistent across an entire field of data, while array encodings are used for
+individual pages of data within a field.  Array encodings can nest other array
+encodings (e.g. a dictionary encoding can bitpack the indices) however array encodings
+cannot nest field encodings.  For this reason data types such as
+``Dictionary<UInt8, List<String>>`` are not yet supported (since there is no dictionary
+field encoding)
+
+.. list-table:: Encodings Available
+   :widths: 15 15 40 15 15
+   :header-rows: 1
 
-    +---------------+----------------+
-    | 0 - 3 byte    | 4 - 7 byte     |
-    +===============+================+
-    | metadata position (uint64)     |
-    +---------------+----------------+
-    | major version | minor version  |
-    +---------------+----------------+
-    |   Magic number "LANC"          |
-    +--------------------------------+
+   * - Encoding Name
+     - Encoding Type
+     - What it does
+     - Supported Versions
+     - When it is applied
+   * - Basic struct
+     - Field encoding
+     - Encodes non-nullable struct data
+     - >= 2.0
+     - Default encoding for structs
+   * - List
+     - Field encoding
+     - Encodes lists (nullable or non-nullable)
+     - >= 2.0
+     - Default encoding for lists
+   * - Basic Primitive
+     - Field encoding
+     - Encodes primitive data types using separate validity array
+     - >= 2.0
+     - Default encoding for primitive data types
+   * - Value
+     - Array encoding
+     - Encodes a single vector of fixed-width values
+     - >= 2.0
+     - Fallback encoding for fixed-width types
+   * - Binary
+     - Array encoding
+     - Encodes a single vector of variable-width data
+     - >= 2.0
+     - Fallback encoding for variable-width types
+   * - Dictionary
+     - Array encoding
+     - Encodes data using a dictionary array and an indices array which is useful for large data types with few unique values
+     - >= 2.0
+     - Used on string pages with fewer than 100 unique elements
+   * - Packed struct
+     - Array encoding
+     - Encodes a struct with fixed-width fields in a row-major format making random access more efficient
+     - >= 2.0
+     - Only used on struct types if the field metadata attribute ``"packed"`` is set to ``"true"``
+   * - Fsst
+     - Array encoding
+     - Compresses binary data by identifying common substrings (of 8 bytes or less) and encoding them as symbols
+     - >= 2.1
+     - Used on string pages that are not dictionary encoded
+   * - Bitpacking
+     - Array encoding
+     - Encodes a single vector of fixed-width values using bitpacking which is useful for integral types that do not span the full range of values
+     - >= 2.1
+     - Used on integral types
```

Comment:
Once things stabilize I think we can replace this with a more formal "spec" where we layout exactly how these encodings work but I don't see any reason to rush at the moment.